src/ferny/session.py

```diff
@@ -141,7 +141,7 @@ async def connect(self,
                 '-o', 'StrictHostKeyChecking=yes',
             ])
 
-        agent = InteractionAgent(interaction_responder)
+        agent = InteractionAgent([interaction_responder] if interaction_responder is not None else [])
```

Comment:
No veto, just a thought: Would a `*handlers` argument look nicer, instead of passing it as a list?

Reply:
I have a strong preference against this practice.  For example, 
```python3
    subprocess.run(['ssh', 'foo'])
```

 takes its arguments as a sequence and the asyncio variants take them as a spread:

```python3
    await asyncio.create_subprocess_exec('ssh', 'foo')
```

I think the `subprocess.run()` variant is much nicer.